schema/schema.go

```diff
@@ -121,8 +122,17 @@ func (s Schema) Prepare(ctx context.Context, payload map[string]interface{}, ori
 			// Handle prepare on an updated document (original provided).
 			oValue, oFound := (*original)[field]
 			// Apply value to change-set only if the field was not identical same in the original doc.
-			if found && (!oFound || !reflect.DeepEqual(value, oValue)) {
-				changes[field] = value
+			if found {
+				if def.Validator != nil {
+					var err error
+					if value, err = def.Validator.Validate(value); err != nil {
+						addFieldError(errs, field, err.Error())
+						continue
+					}
+				}
+				if !oFound || !reflect.DeepEqual(value, oValue) {
+					changes[field] = value
+				}
```

Comment:
Here; treat errors as a s change 😉 

```go
if val := def.Validator; found && val != nil {
	// We treat a validation error as a change; the validation error
	// indicate invalid payload and will be caught again by Validate().
	if validated, err := val.Validate(value); err != nil || !oFound || !reflect.DeepEqual(validated,  oValue) {
		changes[field] = validated
	}
} else if found {
	if refllect.DeepEqual(value, oValue) {
		changes[field] = value
	}
} else if oFound && replace {
	// When replace arg is true ...
	...
}
```

Reply:
I think this snipped is wrong, stating that if values are the same, then it is a change:
```
} else if found {
	if refllect.DeepEqual(value, oValue) {
		changes[field] = value
	}
```
Perhaps we need to use the same expression as above:
```
} else if !oFound || !reflect.DeepEqual(value,  oValue) {
		changes[field] = value
	}
```